### PR TITLE
feat: Add ContentType to headers sent to AWS S3 / DOS

### PR DESCRIPTION
**Pull request by kldavis4**
Based on https://github.com/tinacms/tinacms/pull/4395

> ContentType header was missing in `next-tinacms-s3` package, making AWS serve SVGs as `application/octet-stream`.

This PR solves it by pulling the `mimetype` from the file and if it's not available it fallbacks to `application/octet-stream`

Also makes corresponding change to DOS implementation

### CODE REVIEW

**Review by JackDevAU — APPROVED** [review]

File: packages/next-tinacms-dos/src/handlers.ts [hunk 1]
```diff
@@ -109,6 +109,9 @@ async function uploadMedia(
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
   // @ts-ignore
   const filePath = req.file.path
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const fileType = req.file.mimetype
```
Comment (JackDevAU) [inline packages/next-tinacms-dos/src/handlers.ts:114]:
```suggestion
  const fileType = req.file?.mimetype
```

Does this need a `?`

File: packages/next-tinacms-s3/src/handlers.ts [hunk 2]
```diff
@@ -111,6 +111,9 @@ async function uploadMedia(
     // eslint-disable-next-line @typescript-eslint/ban-ts-comment
     // @ts-ignore
     const filePath = req.file.path
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    const fileType = req.file.mimetype
```
Comment (JackDevAU) [inline packages/next-tinacms-s3/src/handlers.ts:116]: same as above

Comment (kldavis4) [inline packages/next-tinacms-s3/src/handlers.ts:116]:
```suggestion
    const fileType = req.file?.mimetype
```

**Review by JackDevAU — APPROVED** [review]
🚀

**Pull request merged**